### PR TITLE
feat: setting timeout for postgrest-py client

### LINKED ISSUE #225: Setting timeout for postgrest-py client
postgrest-py accepts a timeout parameter from [v0.8.0](https://github.com/supabase-community/postgrest-py/commit/1ea965a6cb32dacb5f41cd1198f8a970a24731b6) 

We don't have an option to set this from supabase-py. So, if the queries run longer than 5s it throws an exception.

### PR DESCRIPTION
**Pull request by keosariel**
postgrest-py accepts a timeout parameter from [v0.8.0](https://github.com/supabase-community/postgrest-py/commit/1ea965a6cb32dacb5f41cd1198f8a970a24731b6)

Added the`timeout` option for the postgrest client and made it optional, with a defaulf of 5sec.

@anand2312 @J0 @dreinon @leynier 
Closes #225

### CODE REVIEW

**Comment by anand2312** [comment]
Closes #225

File: supabase/client.py
Comment (anand2312) [inline supabase/client.py:9]: Can you move this import to the second group of imports? [PEP8](https://peps.python.org/pep-0008/#imports) recommends grouping imports as
1. stdlib
2. related libraries
3. local library imports
Please make this change in the other files as well

File: supabase/client.py
Comment (anand2312) [inline supabase/client.py:160]: [httpx source doesn't seem to include int as an expected timeout type](https://github.com/encode/httpx/blob/67e9c14952340bfd799a0de41dfaf269f376653d/httpx/_types.py#L63) -- are you sure ints work fine here?

**Review by anand2312 — CHANGES_REQUESTED** [review]

**Comment by keosariel** [comment]
> Closes #225 Edit: Whoops that didnt't work as I expected -- can you edit your PR description and add `Closes #225 ` to it so that the issue is linked with this PR please?

okay sure!

File: supabase/client.py
Comment (keosariel) [reply supabase/client.py:9]: Okay would do  that now

File: supabase/client.py
Comment (keosariel) [reply supabase/client.py:160]: Yeah it does, and besides it sort of mirrors the `postgrest`'s code [here ](https://github.com/supabase-community/postgrest-py/blob/09142ab374f51a5ff7b339f071fc8d369c14f144/postgrest/_sync/client.py#L27)

File: supabase/lib/client_options.py
Comment (anand2312) [inline supabase/lib/client_options.py:6]: Nitpicking but imports should be sorted alphabetically :p
```suggestion
from gotrue import SyncMemoryStorage, SyncSupportedStorage
from httpx import Timeout
from postgrest.constants import DEFAULT_POSTGREST_CLIENT_TIMEOUT
```
Apply this for the other file as well, thanks!

**Review by anand2312 — CHANGES_REQUESTED** [review]

File: supabase/lib/client_options.py
Comment (keosariel) [reply supabase/lib/client_options.py]: Ohh my bad. Done ✅

**Comment by keosariel** [comment]
I this getting merged?? @anand2312

**Review by J0 — APPROVED** [review]
Looks good to me -- thanks for the PR! Let's wait for @anand2312 to take a look again before we merge

**Comment by mohnish7** [comment]
Is there an update on this PR? Alternatively how can I adjust the timeout for the client in the mean time? I've tried the methods prescribed [here](https://supabase.com/docs/guides/database/timeouts), but I'm still hitting a 5s timeout (it's preventing me from querying any data from a large view)

**Comment by J0** [comment]
Hey could we trouble you to run `isort` And `black` so the tests pass as well?

Thanks!

**Comment by keosariel** [comment]
> Is there an update on this PR? Alternatively how can I adjust the timeout for the client in the mean time? I've tried the methods prescribed [here](https://supabase.com/docs/guides/database/timeouts), but I'm still hitting a 5s timeout (it's preventing me from querying any data from a large view)

Hey man, If there is any issue then it coming from the `postgrest-py` lib. All I have done is [updated the code](https://github.com/supabase-community/supabase-py/pull/234/files#diff-9c3e4887a114769063f265e09a301bcb16ac543eff2134b8efae3e675a11fd8bR160) to pass a timeout argument to the `postgrest-py` client like you'd [see here](https://github.com/supabase-community/postgrest-py/commit/1ea965a6cb32dacb5f41cd1198f8a970a24731b6#diff-a48a3eadf05dd9e74a42979be983f5c0f9aa74cd05523c63c9d713823d60e398R34)

**Comment by J0** [comment]
Hey @keosariel,

Thanks for the PR! Appreciate the contribution :) We've merged @mohnish7 's extension of this pull and will be closing this for now. 

Let us know if you have any further questions/issues

**Pull request closed without merging**
